src/wallet/asyncrpcoperation_saplingmigration.cpp

```diff
@@ -201,27 +202,10 @@ libzcash::SaplingPaymentAddress AsyncRPCOperation_saplingmigration::getMigration
         assert(saplingAddress != nullptr); // This is checked in init.cpp
         return *saplingAddress;
     }
-    // Derive the address for Sapling account 0
-    auto m = libzcash::SaplingExtendedSpendingKey::Master(seed);
-    uint32_t bip44CoinType = Params().BIP44CoinType();
 
-    // We use a fixed keypath scheme of m/32'/coin_type'/account'
-    // Derive m/32'
-    auto m_32h = m.Derive(32 | ZIP32_HARDENED_KEY_LIMIT);
-    // Derive m/32'/coin_type'
-    auto m_32h_cth = m_32h.Derive(bip44CoinType | ZIP32_HARDENED_KEY_LIMIT);
-
-    // Derive m/32'/coin_type'/0'
-    libzcash::SaplingExtendedSpendingKey xsk = m_32h_cth.Derive(0 | ZIP32_HARDENED_KEY_LIMIT);
-
-    libzcash::SaplingPaymentAddress toAddress = xsk.DefaultAddress();
-
-    if (!HaveSpendingKeyForPaymentAddress(pwalletMain)(toAddress)) {
-        // Sapling account 0 must be the first address returned by GenerateNewSaplingZKey
-        assert(pwalletMain->GenerateNewSaplingZKey() == toAddress);
-    }
-
-    return toAddress;
+    // TODO: move off of legacy addresses.
```

Comment:
Does this contradict the guarantee implied by the release notes that moving all existing funds should make it sufficient to back up the emergency recovery phrase? I would expect migration to count as moving existing funds.

Reply:
No; legacy addresses are generated from the mnemonic seed under the legacy account identifier. Ideally they'd be generated under account ID 0, but this will be more readily implemented once the unified address changeover is complete. There's also the open question as to whether we want to continue to provide functionality to migrate to Sapling at all, or whether we'll change the migration tools to target Orchard instead, so I think that this is an acceptable interim solution.